.devops/code-review-pipelines.yml

```diff
@@ -1,59 +1,105 @@
-    displayName: 'Publish Code Coverage on Azure Devops'     
\ No newline at end of file
+resources:
+  repositories:
+    - repository: pagopaEcommerceLocal
+      type: github
+      name: pagopa/pagopa-ecommerce-local 
+      ref: minor-fix
+      endpoint: 'io-azure-devops-github-ro'
+
+stages:
+  - stage: BuildEndUnitTest
+    jobs: 
+      - job: make_buildEndUnitTest 
+        steps:
+        - task: Cache@2
+          inputs:
+            key: 'maven | "$(Agent.OS)" | pom.xml'
+            restoreKeys: |
+              maven | "$(Agent.OS)"
+              maven
+            path: $(MAVEN_CACHE_FOLDER)
+          displayName: Cache Maven local repo
+
+        - task: SonarCloudPrepare@1
+          displayName: 'Prepare SonarCloud analysis configuration'
+          inputs:
+            SonarCloud: '$(SONARCLOUD_SERVICE_CONN)'
+            organization: '$(SONARCLOUD_ORG)'
+            scannerMode: Other
+            extraProperties: |
+              sonar.projectKey=$(SONARCLOUD_PROJECT_KEY)
+              sonar.projectName=$(SONARCLOUD_PROJECT_NAME)
+              sonar.coverage.exclusions=**/config/*,**/*Mock*,**/model/*,**/utils/soap/*,**/mdcutilities/*,**/PagopaEcommerceTransactionsApplication.java
+              sonar.coverage.jacoco.xmlReportPaths=./target/site/jacoco/jacoco.xml
+              sonar.junit.reportPaths=target/surefire-reports/
+
+        - bash: |
+              docker run -p 10000:10000 -p 10001:10001 -p 10002:10002 mcr.microsoft.com/azure-storage/azurite &
+          displayName: Run azurite to test Service - Storage integration
+
+        - task: Maven@3
+          inputs:
+            mavenPomFile: 'pom.xml'
+            mavenOptions: '-Xmx3072m $(MAVEN_OPTS)'
+            javaHomeOption: 'JDKVersion'
+            jdkVersionOption: '17'
+            jdkArchitectureOption: 'x64'
+            publishJUnitResults: false
+            testResultsFiles: '**/surefire-reports/TEST-*.xml'
+            goals: 'clean verify'
+            sonarQubeRunAnalysis: true
+
+        - task: SonarCloudPublish@1
+          displayName: 'Publish SonarCloud results on build summary'
+          inputs:
+            pollingTimeoutSec: '300'
+
+        - task: PublishCodeCoverageResults@1
+          inputs:
+            codeCoverageTool: 'JaCoCo'
+            summaryFileLocation: 'target/site/jacoco/jacoco.xml'
+            reportDirectory: 'target/site/jacoco'
+          displayName: 'Publish Code Coverage on Azure Devops'
+
+  - stage: IntegrationTest
+    dependsOn: []
+    jobs: 
+      - job: make_integration_test  
+        steps:
+        - checkout: pagopaEcommerceLocal
+        - script: | 
+            echo "##[debug] Change placeholder for ecommerce-transaction-service ..."
+            sed -i.bak  "s/\${ECOMMERCE_TRANSACTIONS_COMMIT_SHA}/$(BRANCH_NAME)/" global-transaction-service.env 
+            cp global-transaction-service.env .env 
+            rm global-transaction-service.env.bak 
```

Comment:
Nitpick: why create a `.bak` file? I think
```bash
sed "s/\${ECOMMERCE_TRANSACTIONS_COMMIT_SHA}/$(BRANCH_NAME)/" global-transaction-service.env > .env
```
Should be sufficient 😃

Reply:
I thank you for the advice i modified the code.